evalml/pipelines/components/transformers/encoders/onehot_encoder.py

```diff
@@ -180,10 +178,37 @@ def categories(self, feature_name):
             raise ValueError(f'Feature "{feature_name}" was not provided to one-hot encoder as a training feature')
         return self._encoder.categories_[index]
 
+    @staticmethod
+    def _make_name_unique(name, seen_before):
+        """Helper to make the name unique."""
+        i = 1
+        while name in seen_before:
+            name = f"{name}_{i}"
```

Comment:
Could we do something like 
```python
i = 1
name = f"{name}_{i}"
while name in seen_before:
    i += 1
    name = f"{name[:name.rindex("_")]}_{i}"
```
It's messier, but I think it makes it cleaner from a user standpoint to see
```
transformed.columns = ['X_y_1', 'X_y_2', 'X_y_3']
```
versus 
```
transformed.columns = ['X_y_1', 'X_y_1_2', 'X_y_1_2_3']
```

Reply:
Great suggestion! I am not sure there is a perfect solution here but I think this is nicer than what I originally had hehe